File: server/src/main/java/de/ingrid/igeserver/documenttypes/UvpType.java
package de.ingrid.igeserver.documenttypes;

import com.orientechnologies.orient.core.db.ODatabaseSession;
import com.orientechnologies.orient.core.metadata.schema.OClass;
import com.orientechnologies.orient.core.metadata.schema.OSchema;
import com.orientechnologies.orient.core.metadata.schema.OType;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.springframework.stereotype.Service;

@Service
public class UvpType extends DocumentType {

    private static Logger log = LogManager.getLogger(UvpType.class);

    private static final String TYPE = "UvpDoc";

    private static final String[] profiles = new String[]{"uvp"};

    public UvpType() {
        super(TYPE, profiles);
    }

    @Override
    public void initialize(ODatabaseSession session) {

        OSchema schema = session.getMetadata().getSchema();
        if (!schema.existsClass(TYPE)) {
            log.debug("Create class " + TYPE);
            OClass addressClass = schema.createClass(TYPE);
            addressClass.createProperty("_id", OType.STRING);
            addressClass.createProperty("_parent", OType.STRING);
        }

    }

}

File: server/src/main/java/de/ingrid/igeserver/documenttypes/UvpType.kt
package de.ingrid.igeserver.documenttypes

import com.orientechnologies.orient.core.db.ODatabaseSession
import com.orientechnologies.orient.core.metadata.schema.OType
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.kotlin.logger
import org.springframework.stereotype.Service

@Service
class UvpType : DocumentType(TYPE, profiles) {

    private val log = logger()

    companion object {
        private const val TYPE = "UvpDoc"
        private val profiles = arrayOf("uvp")
    }

    override fun initialize(session: ODatabaseSession) {
        val schema = session.metadata.schema
        if (!schema.existsClass(TYPE)) {
            log.debug("Create class $TYPE")
            val addressClass = schema.createClass(TYPE)
            addressClass.createProperty("_id", OType.STRING)
            addressClass.createProperty("_parent", OType.STRING)
        }
    }
}